lib/facter/simplib__numa.rb

```diff
@@ -0,0 +1,78 @@
+# _Description_
+#
+# Return true if system has multiple NUMA nodes
+#
+Facter.add("simplib__numa_enabled") do
+  confine :kernel => 'Linux'
+  confine { File.exist? '/sys/devices/system/node' }
+
+  setcode do
+    true
+  end
+
+end
+
+# _Description_
+#
+# Return supported NUMA possible nodes
+#
+Facter.add("simplib__numa_possible_nodes") do
+  confine :kernel => 'Linux'
+  confine { File.exist? '/sys/devices/system/node/possible' }
+
+  # read the file
+  File.open('/sys/devices/system/node/possible', 'r') do | text |
+    result = text.read()
+  end
+
+  setcode do
+    result
+  end
+
+end
+
+# _Description_
+#
+# Return supported NUMA online nodes
+#
+Facter.add("simplib__numa_online_nodes") do
+  confine :kernel => 'Linux'
+  confine { File.exist? '/sys/devices/system/node/online' }
+
+  # read the file
+  File.open('/sys/devices/system/node/online', 'r') do | text |
+    result = text.read()
+  end
+
+  setcode do
+    result
+  end
+
+end
+
+# _Description_
+#
+# Return NUMA memory per node
+#
+Facter.add("simplib__numa_node_memory") do
+  confine :kernel => 'Linux'
+  confine { '/sys/devices/system/node/possible' }
+
+  result = {}
+
+  Dir.glob('/sys/devices/system/node/node*').each do | file |
+    begin
+      File.foreach(file + '/meminfo') do | text |
+        if text =~ / MemTotal:/
+          kb = text.split(':')[1].tr("^0-9", '')
+          bytes = kb * 1024
```

Comment:
I don't think this is what you want.

The result of `kb` will be a String. A String * Integer == that String repeated Integer times.

You probably want `kb.to_i * 1024`.